Add labels param to Google MLEngine Operators

coopergillan opened the pull request:
Pass the `labels` parameter to the training jobs for two Google Machine Learning Operators. This will bring each of them up to snuff with `BigQueryExecuteQueryOperator`, which already accepts this parameter.

closes: #10165

coopergillan committed:
Add labels param to MLEngineStartTrainingJobOperator

Allow MLEngineStartTrainingJobOperator to receive the labels parameter
to be included in the training job. This functionality is already
present in BigQueryExecuteQueryOperator.

coopergillan committed:
Add labels param to MLEngineStartBatchPredictionJobOperator

Allow MLEngineStartBatchPredictionJobOperator to recieve the labels
to then be included in the training job. This functionality is already
present in BigQueryExecuteQueryOperator.

potiuk reviewed (commented):
Would you mind extending the example dags a bit and include the labels? Then they will automatically land in the "Howto Guide",  https://github.com/apache/airflow/blob/master/airflow/providers/google/cloud/example_dags/example_mlengine.py

coopergillan committed:
Add labels to example_dags/example_mlengine.py

mik-laj commented on airflow/providers/google/cloud/operators/mlengine.py:
```diff
@@ -1057,6 +1069,9 @@ def execute(self, context):
             return
 
+        if self._labels:
+            training_request['trainingInput']['labels'] = self._labels
```
I have the impression that the labels should be in ``training_request`, not ``trainingInput``. From what I can see, the API reference documentation for this product is incomplete and the description for the training_input field is missing, 
https://cloud.google.com/ai-platform/prediction/docs/reference/rest/v1/projects.jobs#Job
so you have to look at the Discovery documentation which gets updated more frequently so you have to look at the Discovery documentation, which gets more frequent updates, but also has bugs and missing parameter descriptions.
https://ml.googleapis.com/$discovery/rest?version=v1
More info about Discovery docs: https://developers.google.com/discovery

<img width="773" alt="Screenshot 2020-08-07 at 17 24 00" src="https://user-images.githubusercontent.com/12058428/89661304-cd75c280-d8d2-11ea-940a-98c9c1bf04f3.png">

coopergillan replied:
Thanks for the links. I will take a look and come back with some changes.

coopergillan committed:
Send labels as part of prediction request

Send the labels paramter as part of the prediction requests themselves
and not as part of the prediction input.

coopergillan committed:
Improve typing coverage for labels parameter

coopergillan commented on airflow/providers/google/cloud/operators/mlengine.py:
```diff
@@ -183,6 +185,7 @@ def __init__(self,  # pylint: disable=too-many-arguments
                  project_id: Optional[str] = None,
                  gcp_conn_id: str = 'google_cloud_default',
                  delegate_to: Optional[str] = None,
+                 labels: Optional[Dict[str, str]] = None,
```
I'd like to come back through here and update the other `dict` types with more detail like this.
